Bottom: 2b01224bbcb63fbbb485019b222d315c0bc9dbf4
Top:    41a20f81abcbfe100101265f972e151e97c0af1a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 15:55:59 +0000

Revert "DEBUG STUFF - looking into bad polygons"

This reverts commit 292c3e44c385e3ca13127339cbfdeda1d7f6587e.



---

diff --git a/src/polygon.c b/src/polygon.c
index 3b94ae2..c1f78c1 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -695,9 +695,6 @@ ArcPoly (ArcType * a, Coord thick)
   return ArcPolyNoIntersect (a, thick);
 }
 
-void pline_dump (VNODE * v);
-void poly_dump (POLYAREA * p);
-
 POLYAREA *
 LinePoly (LineType * L, Coord thick)
 {
@@ -766,8 +763,6 @@ LinePoly (LineType * L, Coord thick)
   if (!(np = ContourToPoly (contour)))
     return NULL;
 
-  poly_dump (np);
-
   return np;
 }
 
@@ -1131,28 +1126,10 @@ line_sub_callback (const BoxType * b, void *cl)
   if (!(np = LinePoly (line, line->Thickness + line->Clearance)))
     longjmp (info->env, 1);
 
-  if (info->batch_size == 2)
-    {
-      printf ("Batch no %i, input:\n",info->batch_size);
-      poly_dump (info->accumulate);
-      printf ("---- subtract from it:\n");
-      poly_dump (np);
-    }
-
   poly_Boolean_free (info->accumulate, np, &merged, PBO_UNITE);
-
-  if (info->batch_size == 2)
-    {
-      printf ("Ouptut:\n");
-      poly_dump (merged);
-      printf ("-----------\n");
-    }
-
   info->accumulate = merged;
   info->batch_size ++;
 
-  pcb_printf ("Line %mm, %mm - %mm, %mm\n", line->Point1.X, line->Point1.Y, line->Point2.X, line->Point2.Y);
-
   if (info->batch_size == SUBTRACT_LINE_BATCH_SIZE)
     subtract_accumulated (info, polygon);
 
@@ -1224,8 +1201,7 @@ clearPoly (DataType *Data, LayerType *Layer, PolygonType * polygon,
         r +=
           r_search (layer->line_tree, &region, NULL, line_sub_callback,
                     &info);
-//        subtract_accumulated (&info, polygon);
-      polygon->Clipped = info.accumulate;
+        subtract_accumulated (&info, polygon);
         r +=
           r_search (layer->arc_tree, &region, NULL, arc_sub_callback, &info);
 	r +=
@@ -1234,7 +1210,7 @@ clearPoly (DataType *Data, LayerType *Layer, PolygonType * polygon,
       END_LOOP;
       r += r_search (Data->via_tree, &region, NULL, pin_sub_callback, &info);
       r += r_search (Data->pin_tree, &region, NULL, pin_sub_callback, &info);
-//      subtract_accumulated (&info, polygon);
+      subtract_accumulated (&info, polygon);
     }
   polygon->NoHolesValid = 0;
   return r;
diff --git a/src/polygon1.c b/src/polygon1.c
index 82e64dc..950ea0d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -109,12 +109,12 @@ int vect_inters2 (Vector A, Vector B, Vector C, Vector D, Vector S1,
 
 #define error(code)  longjmp(*(e), code)
 
-#define DEBUG_LABEL
-#define DEBUG_ALL_LABELS
-#define DEBUG_JUMP
-#define DEBUG_GATHER
-#define DEBUG_ANGLE
-#define DEBUG
+#undef DEBUG_LABEL
+#undef DEBUG_ALL_LABELS
+#undef DEBUG_JUMP
+#undef DEBUG_GATHER
+#undef DEBUG_ANGLE
+#undef DEBUG
 #ifdef DEBUG
 #define DEBUGP(...) pcb_fprintf(stderr, ## __VA_ARGS__)
 #else
@@ -154,7 +154,7 @@ pline_dump (VNODE * v)
   do
     {
       n = NEXT_VERTEX(v);
-      pcb_fprintf (stderr, "Line [%mn %mn %mn %mn 10 10 \"%s\"]\n",
+      pcb_fprintf (stderr, "Line [%#mS %#mS %#mS %#mS 10 10 \"%s\"]\n",
 	       v->point[0], v->point[1],
 	       n->point[0], n->point[1], theState (v));
     }
@@ -277,9 +277,8 @@ new_descriptor (VNODE * a, char poly, char side)
   l->angle = ang;
   assert (ang >= 0.0 && ang <= 4.0);
 #ifdef DEBUG_ANGLE
-  DEBUGP ("node on %c at %mn, %mn assigned angle %g on side %c\n", poly,
-	  a->point[0], a->point[1], ang * 10000., side);
-  DEBUGP ("That angle would be %f\n", atan2 (v[1], v[0]) * 180. / M_PI);
+  DEBUGP ("node on %c at %#mD assigned angle %g on side %c\n", poly,
+	  a->point[0], a->point[1], ang, side);
 #endif
   return l;
 }
@@ -1183,7 +1182,7 @@ print_labels (PLINE * a)
 
   do
     {
-      DEBUGP ("%mm, %mm -> %mm, %mm labeled %s\n",
+      DEBUGP ("%#mD->%#mD labeled %s\n",
               EDGE_BACKWARD_VERTEX (e)->point[0], EDGE_BACKWARD_VERTEX (e)->point[1],
                EDGE_FORWARD_VERTEX (e)->point[0],  EDGE_FORWARD_VERTEX (e)->point[1], theState (e));
     }
@@ -1687,7 +1686,7 @@ jump (VNODE **curv, DIRECTION *cdir, J_Rule j_rule)
       return TRUE;
     }
 #ifdef DEBUG_JUMP
-  DEBUGP ("jump entering node at %mm, %mm\n", (*curv)->point[0], (*curv)->point[1]);
+  DEBUGP ("jump entering node at %$mD\n", (*curv)->point[0], (*curv)->point[1]);
 #endif
   /* Pick the descriptor of the edge we came into this vertex with, then spin (anti?)clock-wise one edge descriptor */
   if (*cdir == FORW)
@@ -1706,7 +1705,7 @@ jump (VNODE **curv, DIRECTION *cdir, J_Rule j_rule)
 	      (d->side == 'P' && newone == BACKW))
 	    {
 #ifdef DEBUG_JUMP
-	      DEBUGP ("jump leaving node at %mm, %mm\n",
+	      DEBUGP ("jump leaving node at %#mD\n",
 	              EDGE_DIRECTION_VERTEX (e, newone)->point[0], EDGE_DIRECTION_VERTEX (e, newone)->point[1]);
 #endif
 	      *curv = d->parent;
@@ -1746,7 +1745,7 @@ Gather (VNODE *startv, PLINE **result, J_Rule j_rule, DIRECTION initdir)
 	  poly_InclVertex (PREV_VERTEX (&(*result)->head), newn);
 	}
 #ifdef DEBUG_GATHER
-      DEBUGP ("gather vertex at %mm, %mm\n", curv->point[0], curv->point[1]);
+      DEBUGP ("gather vertex at %#mD\n", curv->point[0], curv->point[1]);
 #endif
       /* Now mark the edge as included.  */
       newn = VERTEX_DIRECTION_EDGE (curv, dir);
